### PR TITLE
New Recipe: Filter Confluent audit logs for Splunk ingest

### PR DESCRIPTION
**Pull request by bbejeck**
### Description 

Resolves issue https://github.com/confluentinc/ksqldb-recipes/issues/113

### New recipe checklist

- [x] Full code validated in Confluent Cloud
- [ ] SQL style/syntax consistent to existing recipes
- [x] Validate presentation with `python3 -m mkdocs serve`
- [x] File structure and content follow [template](https://github.com/confluentinc/ksqldb-recipes/tree/main/template)
- [x] Recipe added to [navigation](https://github.com/confluentinc/ksqldb-recipes/blob/main/mkdocs.yml): comment out until phase release
- [ ] SQL style consistent to existing recipes
- [x] Spell check, e.g. using `aspell`
- [ ] Source connector's auto topic naming convention works with the ksqlDB app values for `KAFKA_TOPIC`

### CODE REVIEW

File: docs/real-time-analytics/filter-logs-for-splunk/process.sql [hunk 1]
```diff
@@ -0,0 +1,50 @@
+SET 'auto.offset.reset' = 'earliest';
+
+CREATE STREAM audit_log_events (
+ log_events STRUCT<
```
Comment (mjsax) [inline docs/real-time-analytics/filter-logs-for-splunk/process.sql:4]: As asked above: is this level of nesting correct? Or should it be:
```
CREATE STREAM audit_log_events (
  id VARCHAR,
  source VARCHAR,
  [...]
)
```

**Review by fifthposition — COMMENTED** [review]
Copyedited.

File: docs/real-time-analytics/filter-logs-for-splunk/index.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                authorizationInfo STRUCT<
+                    granted BOOLEAN,
+                    operation VARCHAR,
+                    resourceType VARCHAR,
+                    resourceName VARCHAR,
+                    patternType VARCHAR,
+                    superUserAuthorization BOOLEAN>
+  ...
+```  
+By supplying a schema to the ksqlDB `STREAM`, you are describing the structure of the data to ksqlDB.  The top-level `log_events` `STRUCT` corresponds to the outermost `{` of the JSON structure and gives a name to the entire JSON entry. The name you supply here is arbitrary but it should be meaningful. You'll notice that there are additional nested `STRUCT` fields representing nested JSON objects within the structure.
```
[the remark this answers is not in the document]
Comment (bbejeck) [reply docs/real-time-analytics/filter-logs-for-splunk/index.md]: ack - I thought the outermost element needed to be named - l learned something new - I've fixed it

**Review by ybyzek — CHANGES_REQUESTED** [review]
@bbejeck per convo today, please check first with Chuck, Johnny, or cloud RBAC team whether fully managed Splunk connector can be used?   Because https://www.confluent.io/blog/visualize-logs-for-simplified-security-in-confluent-cloud/ says self-managed only?

**Comment by bbejeck** [comment]
@ybyzek updated to include a note about assumptions of the location of audit logs.

@mjsax I've addressed all comments

File: docs/real-time-analytics/filter-logs-for-splunk/index.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+This recipe demonstrates how to filter audit logs in a Kafka topic and filter them for specific events and forward them to Splunk for indexing via the [Splunk Sink connector](https://docs.confluent.io/cloud/current/connectors/cc-splunk-sink.html#cc-splunk-sink).  The example data used is from [Confluent Cloud](https://www.confluent.io/confluent-cloud/tryfree/?utm_source=github&utm_medium=ksqldb_recipes&utm_campaign=filter_logs_for_splunk) audit logs.
+The stream processing application filters for events involving operations on `topics`, but you can review the [structure of Confluent audit logs](https://docs.confluent.io/platform/current/security/audit-logs/audit-logs-concepts.html#audit-log-content) and extend this solution to filter for any [auditable event](https://docs.confluent.io/platform/current/security/audit-logs/audit-logs-concepts.html#auditable-events).
+
+!!! note "Recipe Considerations"
```
Comment (bbejeck) [inline docs/real-time-analytics/filter-logs-for-splunk/index.md:14]: @ybyzek added this section describing assumptions

File: mkdocs.yml [hunk 4]
```diff
@@ -34,8 +34,10 @@ nav:
     #- Modernize mainframes for real-time: real-time-analytics/mainframe-offload/index.md
     #- Assess the success of a discount promotion: real-time-analytics/discounting/index.md
     #- Real-time logistics for a delivery fleet: real-time-analytics/logistics/index.md
+    #- Filter Confluent audit logs for Splunk: real-time-analytics/filter-logs-for-splunk
 
 markdown_extensions:
+  - admonition
```
Comment (bbejeck) [inline mkdocs.yml:40]: needed to get admonition block to render

File: docs/real-time-analytics/filter-logs-for-splunk/process.sql [hunk 5]
```diff
@@ -0,0 +1,49 @@
+SET 'auto.offset.reset' = 'earliest';
+
+CREATE STREAM audit_log_events (
+            id VARCHAR,
```
Comment (ybyzek) [inline docs/real-time-analytics/filter-logs-for-splunk/process.sql:4]: @bbejeck would it be ok to make the preceding white space smaller, e.g. two spaces only?  (Here and throughout the ksqlDB statements)

File: mkdocs.yml [hunk 6]
```diff
@@ -34,8 +34,10 @@ nav:
     #- Modernize mainframes for real-time: real-time-analytics/mainframe-offload/index.md
     #- Assess the success of a discount promotion: real-time-analytics/discounting/index.md
     #- Real-time logistics for a delivery fleet: real-time-analytics/logistics/index.md
+    #- Filter Confluent audit logs for Splunk: real-time-analytics/filter-logs-for-splunk
```
Comment (ybyzek) [inline mkdocs.yml:37]:
```suggestion
    #- Filter audit logs for Splunk: real-time-analytics/filter-logs-for-splunk
```

**Review by ybyzek — APPROVED** [review]
@bbejeck well done!  Left a few more minor comments, otherwise LGTM

File: docs/real-time-analytics/filter-logs-for-splunk/index.md (hunk 3 above)
Comment (ybyzek) [reply docs/real-time-analytics/filter-logs-for-splunk/index.md:14]: Perfect, thank you!

**Pull request merged**